python/pyspark/sql/functions.py

```diff
@@ -172,7 +173,9 @@ def lit(col: Any) -> Column:
         return col
     elif isinstance(col, list):
         if any(isinstance(c, Column) for c in col):
-            raise ValueError("lit does not allow a column in a list")
+            raise PySparkException(
```

Comment:
I think we should have an error class like `class PySparkValueError(ValueError, PySparkException)` but I guess you're doing this in a followup

Reply:
Yeah, it's right. We should classify the exception more detail per Python built-in exceptions.

Let me do it in follow-up.